File: Excel VM/AST.Implementation.fs
module AST.Implementation
open Project.Util
open Parser.Definition
open Definition

let fromToken (parseTree:Token, args:CommandLineArguments): AST*CommandLineArguments =
  let transform =
    if args.ContainsKey "optimizations" && args.["optimizations"] = "off"
     then Compile.transformFromToken
     else Compile.transformFromToken >> Optimize.TCOPreprocess   // optimize, maybe pass the parameter to allow more control
  transform parseTree, args

File: docs/scripts/Excel VM/AST.Implementation.js
define(["exports", "./AST.Compile", "./AST.Optimize"], function (exports, _AST, _AST2) {
  "use strict";

  Object.defineProperty(exports, "__esModule", {
    value: true
  });
  exports.fromToken = fromToken;

  function fromToken(parseTree, args) {
    var transform = (args.has("optimizations") ? args.get("optimizations") === "off" : false) ? function (e) {
      return (0, _AST.transformFromToken)(e);
    } : function ($var291) {
      return (0, _AST2.all)((0, _AST.transformFromToken)($var291));
    };
    return [transform(parseTree), args];
  }
});